### PR TITLE
add flag for SegmentItem to trigger folding it into parent WeakAnd

### PR DESCRIPTION
**Pull request by arnej27959**
@bratseth please review

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/prelude/query/WeakAndItem.java [hunk 1]
```diff
@@ -87,6 +87,32 @@ protected void encodeThis(ByteBuffer buffer) {
...
+                }
+            } else {
+                result.addItem(child);
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public int encode(ByteBuffer buffer) {
```
Comment (bratseth) [inline container-search/src/main/java/com/yahoo/prelude/query/WeakAndItem.java:105]: super.encode will encode this, then encode all the items in this (without folding). Won't the total result of this be a bit strange?

Comment (arnej27959) [reply container-search/src/main/java/com/yahoo/prelude/query/WeakAndItem.java:105]: super.encode() should be called when no folding is needed; maybe the logic was not clear. will refactor some...

**Comment by arnej27959** [comment]
@bratseth please take another look

**Review by bratseth — APPROVED** [review]
👍

**Pull request merged**